pkg/token/manualtoken_test.go

```diff
@@ -0,0 +1,76 @@
+package token
+
+import (
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+)
+
+func TestNewManualToken(t *testing.T) {
+	oAuthConfig := adal.OAuthConfig{}
+	clientID := "test-client-id"
+	resourceID := "test-resource-id"
+	tenantID := "test-tenant-id"
+	token := &adal.Token{AccessToken: "test-access-token"}
+
+	// Test with valid inputs
```

Comment:
this test code is very repetitive. typically you define the test matrix and use a loop to exercise them and verify the expectations. you can refer to pkg/converter/convert_test.go

Reply:
Kindly review again to check test matrix